src/js/components/DataTable/Header.js

```diff
@@ -18,6 +18,42 @@ import {
   StyledDataTableRow,
 } from './StyledDataTable';
 import { datumValue } from './buildState';
+import { kindPartStyles } from '../../utils';
+
+// build up CSS from basic to specific based on the supplied sub-object paths.
+// adapted from StyledButtonKind to only include parts relevant for DataTable
+const buttonStyle = ({ theme }) => {
+  const styles = [];
+  // don't apply background or border on button. these
+  // should only apply to the `th`
+  const contentStyles = { ...theme.dataTable.header };
+  delete contentStyles.background;
+  delete contentStyles.border;
```

Comment:
Should we also be omitting `color` and `font`, since those are button-y?

Reply:
Handled this with new utility function `separateThemeProps`